opentelemetry-declarative-config/src/lib.rs

```diff
@@ -0,0 +1,353 @@
+//! # OpenTelemetry declarative configuration
+//!
+//! This crate provides a declarative way to configure OpenTelemetry SDKs using YAML files.
+
+pub mod logs;
+pub mod metrics;
+pub mod telemetry_config;
+
+use opentelemetry::global;
+use opentelemetry_sdk::{
+    error::OTelSdkResult, logs::SdkLoggerProvider, metrics::SdkMeterProvider,
+    trace::SdkTracerProvider,
+};
+
+use crate::{
+    logs::{BatchExporterFactory, LogsBatchExporterFactory, LogsConfig},
+    metrics::{
+        MetricsConfig, MetricsPeriodicExporterFactory, MetricsPullExporterFactory,
+        PeriodicExporterFactory, PullExporterFactory,
+    },
+    telemetry_config::TelemetryConfig,
+};
+
+pub struct Configurator {}
+
+impl Configurator {
+    pub fn new() -> Self {
+        Self {}
+    }
+
+    pub fn configure_telemetry_from_yaml(
+        &self,
+        telemetry_config_str: String,
+    ) -> Result<TelemetryProviders, Box<dyn std::error::Error>> {
+        let config = TelemetryConfig::from_yaml(&telemetry_config_str)?;
+        self.configure_telemetry(config)
+    }
+
+    pub fn configure_telemetry_from_yaml_file(
+        &self,
+        file_path: &str,
+    ) -> Result<TelemetryProviders, Box<dyn std::error::Error>> {
+        let config = TelemetryConfig::from_yaml_file(file_path)?;
+        self.configure_telemetry(config)
+    }
+
+    pub fn configure_telemetry(
+        &self,
+        telemetry_config: TelemetryConfig,
+    ) -> Result<TelemetryProviders, Box<dyn std::error::Error>> {
```

Comment:
Not for this iteration, but good to see if we can return more structured errors to the caller.